Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm

```diff
@@ -3351,40 +3351,81 @@ - (CABasicAnimation *)_animationForFindOverlay:(BOOL)animateIn
+    auto rightView = _findOverlaysForOverflowScroll[FindOverlayView::Right].get();
+    auto bottomView = _findOverlaysForOverflowScroll[FindOverlayView::Bottom].get();
+    auto leftView = _findOverlaysForOverflowScroll[FindOverlayView::Left].get();
+
+    CGRect contentViewBounds = [_contentView bounds];
+    CGRect contentViewFrame = [_contentView frame];
+    CGFloat minX = std::min<CGFloat>(0, scrollView.contentOffset.x);
+    CGFloat minY = std::min<CGFloat>(0, scrollView.contentOffset.y);
+    CGFloat maxX = std::max<CGFloat>(scrollView.bounds.size.width + scrollView.contentOffset.x, contentViewBounds.size.width);
+    CGFloat maxY = std::max<CGFloat>(scrollView.bounds.size.height + scrollView.contentOffset.y, contentViewBounds.size.height);
+
+    [topView setFrame:CGRectMake(
+        CGRectGetMinX(contentViewFrame),
+        minY,
+        std::max<CGFloat>(maxX - CGRectGetMinX(contentViewFrame), 0),
+        std::max<CGFloat>(CGRectGetMinY(contentViewFrame) - minY, 0))];
+
+    [rightView setFrame:CGRectMake(
+        CGRectGetMaxX(contentViewFrame),
+        CGRectGetMinY(contentViewFrame),
+        std::max<CGFloat>(maxX - CGRectGetMaxX(contentViewFrame), 0),
+        std::max<CGFloat>(maxY - CGRectGetMinY(contentViewFrame), 0))];
+
+    [bottomView setFrame:CGRectMake(
+        minX,
+        CGRectGetMaxY(contentViewFrame),
+        std::max<CGFloat>(CGRectGetMaxX(contentViewFrame) - minX, 0),
+        std::max<CGFloat>(maxY - CGRectGetMaxY(contentViewFrame), 0))];
+
+    [leftView setFrame:CGRectMake(
+        minX,
+        minY,
+        std::max<CGFloat>(CGRectGetMinX(contentViewFrame) - minX, 0),
+        std::max<CGFloat>(CGRectGetMaxY(contentViewFrame) - minY, 0))];
 }
 
 - (void)_showFindOverlay
 {
-    if (!_findOverlay) {
-        _findOverlay = adoptNS([[UIView alloc] init]);
-        UIColor *overlayColor = [UIColor colorWithRed:(26. / 255) green:(26. / 255) blue:(26. / 255) alpha:(64. / 255)];
-        [_findOverlay setBackgroundColor:overlayColor];
+    UIColor *overlayColor = [UIColor colorWithRed:(26. / 255) green:(26. / 255) blue:(26. / 255) alpha:(64. / 255)];
+    if (!_findOverlaysForOverflowScroll.size()) {
+        for (size_t i = 0 ; i < 4 ; i++) {
+            auto newOverlay = adoptNS([[UIView alloc] init]);
+            [newOverlay setBackgroundColor:overlayColor];
+            [_scrollView insertSubview:newOverlay.get() belowSubview:_contentView.get()];
+            _findOverlaysForOverflowScroll.append(WTFMove(newOverlay));
+        }
     }
-
-    [self _updateFindOverlayPosition];
-    [_scrollView insertSubview:_findOverlay.get() belowSubview:_contentView.get()];
+    [self _updateFindOverlayForOverflowScrollPositions];
 
     if (CALayer *contentViewFindOverlayLayer = [self _layerForFindOverlay]) {
-        [[_findOverlay layer] removeAllAnimations];
-        [contentViewFindOverlayLayer removeAllAnimations];
+        for (RetainPtr overlay : _findOverlaysForOverflowScroll) {
```

Comment:
Given the repetition of this loop, we could have a method like `_updateFindOverlaysOutsideContentView:` that takes a block, and calls the block on each view.

Reply:
or just make it an initializer list like `for (RetainPtr overlay : { _findOverlaysForOverflowScroll.left, _findOverlaysForOverflowScroll.right, etc {)` ?